compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -839,6 +868,48 @@ impl<'a> Parser<'a> {
         Ok((v, trailing, recovered))
     }
 
+    fn recover_missing_braces_around_closure_body(
+        &mut self,
+        right_pipe_span: Span,
+        mut expect_err: DiagnosticBuilder<'_>,
+    ) -> PResult<'a, ()> {
+        let initial_semicolon = self.token.span;
+
+        expect_err.span_help(
+            initial_semicolon,
+            "This `;` turns the expression into a statement, which must be placed in a block",
+        );
+
+        while self.eat(&TokenKind::Semi) {
+            let _ = self.parse_stmt(ForceCollect::Yes)?;
+        }
+
+        expect_err.set_primary_message(
+            "closure body that contain statements must be surrounded by braces",
```

Comment:
There's a grammar problem here: you either want "**a** closure body that contain**s** statements must be surrounded by braces" or ""closure bod**ies** that contain statements must be surrounded by braces". We may have a stylistic convention on which one is correct, but if so, I don't know what it is.

Reply:
That makes sense, thanks. The [suggestion style guide](https://rustc-dev-guide.rust-lang.org/diagnostics.html?#suggestion-style-guide) doesn't help unfortunately. I have found no other wording suggestion in the documentation.

I like your second suggestion and I'll keep that one for no particular reason other than being grammatically correct. Feel free to suggest anything that may be better :)